### ISSUE TITLE
Questions about nerf studio local install

### ISSUE THREAD

**Issue by gitihobo** [nerfstudio-project/nerfstudio#2095]
Does nerf run locally? How does one implement a local version of nerf studio through anaconda env?

**Comment by tancik** [comment]
Yep it can be run locally, the installation instructions are the same.

**Comment by gitihobo** [comment]
I installed nerfstudio, but how do I actually open the app and use it now? Dont understand the tutorial

**Comment by gitihobo** [comment]
Installed pytorch and git cloned and installed dependencies, how do I run the training locally and open the viewer?

**Comment by maturk** [comment]
@gitihobo, you can follow the tutorials here: https://docs.nerf.studio/en/latest/quickstart/first_nerf.html.

**Comment by gitihobo** [comment]
Yeah thats the issue I am having I follow the tutorials to where I understand them and I get errors, I dont know if what I am doing is correct, I Installed pytorch and git cloned and installed dependencies, then I try ns-download-data nerfstudio --capture-name=poster and I get an error

**Comment by maturk** [comment]
> Yeah thats the issue I am having I follow the tutorials to where I understand them and I get errors, I dont know if what I am doing is correct, I Installed pytorch and git cloned and installed dependencies, then I try ns-download-data nerfstudio --capture-name=poster and I get an error

Maybe you can hop on the Discord server: https://discord.gg/uMbNqcraFc. You can post some screenshots with your issues and ask questions on the #troubleshooting forum.

**Comment by gitihobo** [comment]
I made the github issue so other people in my shoes could see the solution, I dont really use discord. but I seem to get a ModuleNotFoundError: No module named "nerfstudio" error, I conda list and it says nerfstudio is present, additional info: to avoid  double folder names I called my environment nerfstudio and placed all the github files there, hopefully that doesn't confuse the directories.

**Comment by maturk** [comment]
> I made the github issue so other people in my shoes could see the solution, I dont really use discord. but I seem to get a ModuleNotFoundError: No module named "nerfstudio" error, I conda list and it says nerfstudio is present, additional info: to avoid double folder names I called my environment nerfstudio and placed all the github files there, hopefully that doesn't confuse the directories.

When you type the command "pip install -e ." do any errors come then? and when you type "pip list" does nerfstudio also come up in the list of python packages? I suspect there is something wrong with the installation due to this no module error.

**Comment by gitihobo** [comment]
I fresh installed and got it to start training, but back to my original question, it seems to need online connection, how can I make this a purely local install for privacy and ownership reasons

**Comment by maturk** [comment]
> I fresh installed and got it to start training, but back to my original question, it seems to need online connection, how can I make this a purely local install for privacy and ownership reasons

Cool. The viewer does not require internet connection if running locally, it runs on the localhost. For example localhost:7007 is the default port for the viewer. For example, this is probably the default web address that the viewer tries to create when using ns-train `https://viewer.nerf.studio/versions/23-05-15-1/?websocket_url=ws://localhost:7007` which is actually just on your localhost.

**Comment by gitihobo** [comment]
Am I misunderstanding? it says To use our viewer, you need to connect to the WebSocket server where your training job is running. Enter your WebSocket URL below and click the link to connect to your training job ws://localhost: doesnt this mean its using the web to train? Also, is it possible to export pieces of the composition as a 3d object? I want to export only the chair so I can use it as an asset for a 3d game

**Comment by maturk** [comment]
> Am I misunderstanding? it says To use our viewer, you need to connect to the WebSocket server where your training job is running. Enter your WebSocket URL below and click the link to connect to your training job ws://localhost: doesnt this mean its using the web to train? Also, is it possible to export pieces of the composition as a 3d object? I want to export only the chair so I can use it as an asset for a 3d game

I am no expert in these web app things, but the Nerfstudio viewer is an application that simply runs in a browser. This does not mean you need internet connection, it just means it uses websockets and modern browser API stuff to run the program. The application is started when you do the "ns-train" or "ns-render" commands. This starts the webapp that uses your computers websocket connections to transfer data from all the python/torch code to something that can be visualized in everyday browsers. The application itself does not require internet connection, because the server that it is actually running on is your own computer when you start the app. This is what is meant by the localhost thing. However, the cool thing is with these websockets is that, if you are on a remote computer, you can use port-forwarding and display the same GUI in another web-browser through the internet. But the server is still the remote computer. Someone with more experience with this can fill in if I am incorrect somewhere.

**Comment by maturk** [comment]
> OK thank you, I guess i'll close this issue soon I just need help with one final thing, I made a video based on the default project is starts with, but I am a noob to this type of stuff, I dont really understand how to put my own images in the software, how do I open nerfstudio and import my images and train?

You will have to use ns-process-data command: https://docs.nerf.studio/en/latest/reference/cli/ns_process_data.html. Check out the good docs here: https://docs.nerf.studio/en/latest/quickstart/custom_dataset.html. You can use Colmap or the other methods listed there. Everything works using command line arguments. The first step in the NeRF pipeline is pose estimation for your images, and this is what the ns-process-data command does for you. If that is successful, then you can ns-train.

**Comment by thzva** [comment]
I have also had this problem recently. 

My solution is : 
1. delete conda environment by rm *rf ( rm -rf /home/***/anaconda3/envs/nerfstudio)
2. conda clean --all
3. sudo rm -r ~/.cache/pip
4. reinstall the nerfstudio conda environment

This issue has been temporarily resolved, but the exact reason is unclear.

**Comment by gasparyanartur** [comment]
> I have also had this problem recently.
> 
> My solution is :
> 
>     1. delete conda environment by rm _rf ( rm -rf /home/_**/anaconda3/envs/nerfstudio)
> 
>     2. conda clean --all
> 
>     3. sudo rm -r ~/.cache/pip
> 
>     4. reinstall the nerfstudio conda environment
> 
> 
> This issue has been temporarily resolved, but the exact reason is unclear.

Thanks, this worked for me!
